astrowidgets/interface_definition.py

```diff
@@ -0,0 +1,255 @@
+from typing import Protocol, runtime_checkable, Any
+from abc import abstractmethod
+
+# Allowed locations for cursor display
+ALLOWED_CURSOR_LOCATIONS = ('top', 'bottom', None)
+
+# List of marker names that are for internal use only
+RESERVED_MARKER_SET_NAMES = ('all',)
+
+__all__ = [
+    'ImageViewerInterface',
+    'ALLOWED_CURSOR_LOCATIONS',
+    'RESERVED_MARKER_SET_NAMES'
+]
+
+
+@runtime_checkable
+class ImageViewerInterface(Protocol):
+    # These are attributes, not methods. The type annotations are there
+    # to make sure Protocol knows they are attributes. Python does not
+    # do any checking at all of these types.
+    click_center: bool
+    click_drag: bool
+    scroll_pan: bool
+    image_width: int
+    image_height: int
+    zoom_level: float
+    is_marking: bool
+    stretch_options: tuple
+    autocut_options: tuple
+    cursor: str
+    marker: Any
```

Comment:
These were omitted the first time through:

    zoom_level: float
    is_marking: bool
    stretch_options: tuple
    autocut_options: tuple
    cursor: str